### PR TITLE
 add drag and drop function and task counter

### PR DESCRIPTION
**Pull request by Malte1983**
Drag and drop function added. Task counter added. Test and story follow. 

### this Resolve
userstory-#3

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -33,24 +36,45 @@ function App() {
     }
   })
 
+  const uncompletedTaks = tasks.filter(task => !task.completed)
+
+  const tasksNoun = tasks.length !== 1 ? '' : ''
```
Comment (StefanBehring) [inline src/App.js:41]: Right now the result of the tenary is always: `''`

File: src/components/Tasklist/ToDoCard.js [hunk 2]
```diff
@@ -30,42 +31,44 @@ export default function ToDoCard({
...
+          </SortableKnob>
+          {updateTask ? (
+            <Section>
+              <label htmlFor="edit" />
+              <EditInputField
+                onChange={handleChange}
+                type="text"
+                name="edit"
+                value={value}
+              ></EditInputField>
```
Comment (StefanBehring) [inline src/components/Tasklist/ToDoCard.js:50]: You can write it like this:
```
<EditInputField
                onChange={handleChange}
                type="text"
                name="edit"
                value={value}
              />
```

File: src/components/Tasklist/ToDoCard.js [hunk 3]
```diff
@@ -30,42 +31,44 @@ export default function ToDoCard({
...
+          <Checkbox
+            type="checkbox"
+            checked={completed}
+            onChange={() => onHandleIsChecked(id)}
+          />
+          <ButtonEdit onClick={updateTaskHandler}>
+            <img src={edit} alt="menu" width="30" />
+          </ButtonEdit>
+          <ButtonTrash onClick={() => onHandleDeleteTask(id)}>
+            <img src={trashIcon} alt="menu" width="25" />
```
Comment (StefanBehring) [inline src/components/Tasklist/ToDoCard.js:67]: You have the same alt text on every img. I guess you should change them to indicate what the image is used for or what it shows.

**Comment by StefanBehring** [comment]
Hi, I added a few suggestions!
But: There is some really nice code in it! Really well done!

File: src/components/Tasklist/ToDoCard.js [hunk 4]
```diff
@@ -3,7 +3,8 @@ import styled from 'styled-components/macro'
 import lines from '../../assets/lines.svg'
 import trashIcon from '../../assets/trashIcon.svg'
 import edit from '../../assets/edit.svg'
-import done from '../../assets/done.svg'
+import donegreen from '../../assets/donegreen.svg'
```
[the remark this answers is not in the document]
Comment (Malte1983) [reply src/components/Tasklist/ToDoCard.js:6]: I have adjusted the file name and the corresponding paths.

File: src/components/Tasklist/ToDoCard.js (hunk 2 above)
Comment (Malte1983) [reply src/components/Tasklist/ToDoCard.js:50]: You are absolutely right, this makes the code more readable. I have adjusted it.

File: src/components/Tasklist/ToDoCard.js (hunk 3 above)
Comment (Malte1983) [reply src/components/Tasklist/ToDoCard.js]: rename  alt text for each image  :)

**Pull request merged**